spec/controllers/selfservice/principals_controller_spec.rb

```diff
@@ -0,0 +1,51 @@
+RSpec.describe SelfService::PrincipalsController, type: :controller do
+  before { sign_in(user) }
+  let(:user) { FactoryGirl.create(:user, principal: principal) }
+  let(:principal) { FactoryGirl.create(:principal) }
+
+  describe 'GET #edit' do
+    before { get :edit, id: principal.id }
+
+    it 'renders the edit action' do
+      expect(subject).to render_template 'self_service/principals/edit'
+    end
+
+    it 'provides the principal object' do
+      expect(assigns(:principal)).to eq principal
+    end
+  end
+
+  describe 'PATCH #update' do
+    before { patch :update, id: principal.id, principal: principal_params }
+
+    context 'with valid parameters' do
+      let(:principal_params) { FactoryGirl.attributes_for(:principal) }
+
+      it 'saves the record' do
+        expect(assigns(:principal)).to be_persisted
+        expect(assigns(:principal).first_name).to eq principal_params[:first_name]
+        expect(assigns(:principal).last_name).to eq principal_params[:last_name]
+        expect(assigns(:principal).email_address).to eq principal_params[:email_address]
+        expect(assigns(:principal).website_address).to eq principal_params[:website_address]
+        expect(assigns(:principal).job_title).to eq principal_params[:job_title]
+        expect(assigns(:principal).telephone_number).to eq principal_params[:telephone_number]
+      end
+
+      it 'redirects to the edit action' do
+        expect(response).to redirect_to edit_self_service_principal_path(principal)
+      end
+    end
+
+    context 'with invalid paramters' do
+      let(:principal_params) { { email_address: 'no' } }
+
+      it 'does not save the record' do
+        expect(assigns(:principal)).to be_changed
+      end
+
```

Comment:
Mm, I suppose it does read counter-intuitively. `changed?` is defined [here](http://api.rubyonrails.org/classes/ActiveModel/Dirty.html#method-i-changed-3F). So I'm testing that the object has unsaved changes on it. But it could read better.